api/src/main/java/org/pmiops/workbench/exfiltration/EgressLogService.java

```diff
@@ -139,13 +150,31 @@ private AuditEgressRuntimeLogEntry toRuntimeLogEntry(FieldValueList fvl) {
         .message(fvl.get("message").getStringValue());
   }
 
+  private String getDatasetId(Optional<SumologicEgressEvent> sumologicEgressEvent) {
+    // If getGceEgressMib is more than zero, we see this egress triggered by GCE, otherwise, it's
+    // Dataproc.
+    // TODO(yonghao): Revisit this logic and probably make an enum once we understand what does GKE
+    // sumologic event looks like.
+    boolean isGce =
+        sumologicEgressEvent
+            .map(SumologicEgressEvent::getGceEgressMib)
+            .map(d -> d > 0)
+            .orElse(false);
+    return isGce
+        ? workbenchConfigProvider.get().firecloud.workspaceLogsProject + GCE_LOG_LOCATION
+        : workbenchConfigProvider.get().firecloud.workspaceLogsProject + DATAPROC_LOG_LOCATION;
+  }
+
   private Job startBigQueryJob(
-      EgressTerraRuntimeLogPattern runtimeLogPattern, Map<String, QueryParameterValue> baseParams) {
+      EgressTerraRuntimeLogPattern runtimeLogPattern,
+      Map<String, QueryParameterValue> baseParams,
+      String datasetId) {
     return bigQueryService.startQuery(
         QueryJobConfiguration.newBuilder(
                 String.format(
                     "SELECT timestamp, jsonPayload.message AS message"
-                        + " FROM `%s.WorkspaceRuntimeLogs.cos_containers`"
+                        + " FROM "
```

Comment:
nit: we're using `String.format()` here, so this can be `" FROM %s "` if we update the param